crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -39,7 +39,10 @@ impl<I, O> RemovedSystem<I, O> {
 /// These are opaque identifiers, keyed to a specific [`World`],
 /// and are created via [`World::register_system`].
 #[derive(Eq)]
-pub struct SystemId<I = (), O = ()>(Entity, std::marker::PhantomData<fn(I) -> O>);
+pub struct SystemId<I = (), O = ()>(
```

Comment:
I'd prefer to swap this away from a tuple struct while we're at it.

Reply:
Turning SystemId into a regular struct?

Reply:
Yes please.